content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,126 @@
++++
+path = "inside-rust/9999/12/31/redesigning-the-initial-bootstrap-sequence"
+title = "Redesigning the Initial Bootstrap Sequence"
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "the Bootstrap team"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+This blog post accompanies an [upcoming major change to the `bootstrap` build system][stage0-redesign-pr] (see also [Major Change Proposal 619][redesign-stage0-mcp]), which is expected to affect contributors who works on the `rustc` compiler, the standard library, tools, and the distributors of rust toolchains.
+
+# TL;DR: What is being changed?
+
+We are [redesigning the stage 0 bootstrap sequence][stage0-redesign-pr] so that the standard library will now only support one compiler version. Building the stage 1 compiler will no longer involve building the in-tree standard library. Instead, the pre-built beta standard library will be used.
+
+The following section is a quick primer on the concept of bootstrapping and the terminology we use in this blog post.
+
+# A quick primer on bootstrapping and terminology used in this blog post {#preliminary}
+
+This section is intended to explain some basic bootstrapping concepts to make it easier for contributors to understand the upcoming bootstrap changes. You can [skip this section](#motivation) if you are already familiar with [the `bootstrap` build system][bootstrap] itself.
+
+## What is bootstrapping? {#intro-to-bootstrapping}
+
+[*Bootstrapping*][compiler-bootstrapping] is the process of producing a newer version of a compiler with an older version of itself. Usually, bootstrap processes involve the concept of *stages*, where a compiler and associated artifacts from a stage is used to build the compiler of the next stage. For example, a stage 2 compiler is built from a stage 1 compiler and associated stage 1 artifacts. The stage number is named after the compiler in question.
+
+## Bootstrapping `rustc` {#bootstrapping-rust}
+
+C compilers like [`gcc`][gcc-bootstrap] or [`clang`][clang-bootstrap] do not build their standard libraries from source, but instead links against the same specified standard library across all bootstrap stages. Typically, this specified standard library is the system standard library.[^adapted]
+
+In the context of `rustc`, this is not the case, as the compiler and standard library are tightly coupled through *intrinsics* and *lang items*. Bootstrapping `rustc` involves building the standard library from source. Building a stage `N` `rustc` requires building or acquiring the stage `N - 1` `rustc`, as well as the stage `N - 1` standard library built by the stage `N - 1` `rustc`.
```

Comment:
It feels a little misleading to say that this is not the case for rustc because X when that X is not changing with this change -- both lang items and intrinsics are here to stay, and they're just as tightly coupled to a compiler version.

In fact, the reality that we *can* make this change implies this is basically an arbitrary choice for Rust: do we prefer cfg(bootstrap) in libstd or in librustc*? (Right?)

Not sure if there's a way to rephrase this to avoid that implication...

Reply:
> In fact, the reality that we can make this change implies this is basically an arbitrary choice for Rust: do we prefer cfg(bootstrap) in libstd or in librustc*? (Right?)

Yeah, this is the tradeoff that we're changing here, to prefer the less-common `cfg(bootstrap)` for compiler instead of the more frquent `cfg(bootstrap)` for std.

Reply:
This was reworded as recommended in (round 8).